### PR TITLE
Instaflux demo media store integration

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR adds a way to create a media post in the Instaflux demo. The UI is temporary as it only adds a button which lets you pick media and it creates a post with that media. It's just a v1 of the implementation.

@tonyr59h I am opening this PR a bit earlier for us to have a place to discuss further implementation. What I understand from the conversations with @maxme is, this will be a pretty simple version of Instagram which uses wordpress as it's backend. So, what I propose is we ditch the create text post implementation completely. The example app already provides an example of that. Instead, after you log in, we should land on a page where we show the current posts in that blog. It could be pretty simple, just the title and the featured image. We can add a button to create a new media post which will open the camera or the media library and we just upload it and create a post around it (to keep things simple) and refresh the post list. This should be good enough of a demo. What do you think?

P.S: Since this is going to be merged to the `demo/instaflux` branch and not `develop`, I think it's best to review it and merge it and keep doing other enhancements on different branches rather than keep working on this.

### CODE REVIEW

**Comment by oguzkocer** [comment]
I forgot to add this in the description. The current  implementation won't work as the media upload event doesn't return the media information. Just like in the example app, it should work once this TODO is resolved:

`// TODO: serialize MediaModel from response and add to resultList`

https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/demo/instaflux-media/fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/media/MediaRestClient.java#L206

**Comment by tonyr59h** [comment]
> So, what I propose is we ditch the create text post implementation completely.

Agreed! Simple is usually better and media was the original intent of the project.

**Comment by tonyr59h** [comment]
> ...What do you think?

I like the design :)

> it should work once this TODO is resolved

It's been resolved on the [media branch](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/feature/39-media-store/fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/media/MediaRestClient.java#L239), I'll merge it into `demo/instaflux`.

File: instaflux/src/main/java/org/wordpress/android/fluxc/instaflux/PostActivity.java [hunk 1]
```diff
@@ -109,6 +201,20 @@ private void signOut() {
         }
     }
 
+    private void uploadMedia(String imagePath, String mimeType) {
+        SiteModel site = mSiteStore.getSites().get(0);
+        MediaModel mediaModel = new MediaModel();
+        mediaModel.setFilePath(imagePath);
+        mediaModel.setFileExtension(imagePath.substring(imagePath.lastIndexOf(".") + 1, imagePath.length()));
```
Comment (tonyr59h) [inline instaflux/src/main/java/org/wordpress/android/fluxc/instaflux/PostActivity.java]: Not necessary, but we can use some [utility methods](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/feature/39-media-store/fluxc/src/main/java/org/wordpress/android/fluxc/utils/MediaUtils.java#L19) here :)

File: instaflux/src/main/java/org/wordpress/android/fluxc/instaflux/PostActivity.java [hunk 2]
```diff
@@ -146,4 +258,18 @@ public void onPostUploaded(PostStore.OnPostUploaded event) {
         mContentText.setText("");
         ToastUtils.showToast(this, event.post.getTitle());
     }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onMediaChanged(MediaStore.OnMediaChanged event) {
+        switch (event.causeOfChange) {
+            case UPLOAD_MEDIA:
+                if (event.media.size() > 0) {
```
Comment (tonyr59h) [inline instaflux/src/main/java/org/wordpress/android/fluxc/instaflux/PostActivity.java]: NPE warning: `event.media` could be null here 😈

**Comment by tonyr59h** [comment]
:shipit:

**Pull request merged**